test/validate.jl

```diff
@@ -0,0 +1,14 @@
+@testset "XML Validation with XSD" begin
+  
+    @test validate("valid.xml", "valid.xsd")
+
+    doc = parse_file("valid.xml")
+    schema = XMLSchema("valid.xsd")
+
+    @test validate("valid.xml", schema)
+
+    @test validate(doc, schema)
+
+    @test validate(root(doc), schema)
```

Comment:
Let's add a test to shows validation failure

Reply:
I am uncertain why XSD/XML file pairs that work with the many online XML/XSD validators fail under Linux. It may be a libxml2 quirk. I will see if I can find a XSD/XML pair of files which work under Ubuntu with Julia 1.5.

Reply:
I think the current PR contains the suggestions of @Omus and works except for the FreeBSD build problem with recent Julia versions.